### ISSUE TITLE
Dashes, "smart" quotes  

### ISSUE THREAD

**Issue by ile** [ckeditor/ckeditor5-autoformat#62]
Would this be the correct plugin to handle things like:

- Turn double dash (--) into – (endash) 
- Turn triple dash (---) into — (emdash) 
https://en.wikipedia.org/wiki/Dash

- Turn "..." into “...”  
- Turn '...'  into ‘...’
(or more "beautiful" quotation marks)

- If a text is selected (as in "painted") and then a quotation mark is typed, the **selected** text is wrapped in quotes.

If this would be the correct plugin, let's treat this issue as a feature request? Thank you.

**Comment by scofalik** [comment]
This topic was briefly discussed here: https://github.com/ckeditor/ckeditor5/issues/1366. The short answer is: it probably is possible to implement those transformations using what is already available in autoformat feature but those kinds of transformations weren't on our mind when we created `InlineAutoformatEditing` so I can't guarantee you that it will work flawlessly or that the integration will look good.

**Comment by ile** [comment]
Ok.

It would be nice to have a ckeditor-typography package, which would do something like this:

https://github.com/pnevyk/tipograph
https://github.com/sapegin/richtypo.js

I know I could perhaps create it myself, and maybe I will, but if you did it, it would probably be correctly done from the beginning...

**Comment by Reinmar** [comment]
Agree. We actually planned this package and there might've even been a ticket for it (can't find it though). The autoformatting feature is about formatting, so it's a bit different topic. Text transformations would be a separate thing.

I'm moving this ticket to the main repo and confirming.
